x-pack/test/cloud_security_posture_functional/page_objects/findings_page.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import type { FtrProviderContext } from '../ftr_provider_context';
+
+// Defined in EUI
+const getEuiPaginationText = (size: 10 | 25) => `Rows per page: ${size}`;
+const getEuiPaginationRowTestId = (size: 10 | 25) => `tablePagination-${size}-rows`;
+const getEuiPaginationButtonTestId = (index: number) => `pagination-button-${index}`;
+const EUI_PAGINATION_BUTTON = 'tablePaginationPopoverButton';
+
+// Defined in CSP plugin
+const FINDINGS_ROUTE = 'cloud_security_posture/findings';
+const FINDINGS_TABLE_TESTID = 'findings_table';
+const FILTER_CELL_VALUE_SELECTOR = 'div[data-test-subj="filter_cell_value"]';
+
+// Defined in Security Solution plugin
+const SECURITY_SOLUTION_APP_NAME = 'securitySolution';
+
+export function FindingsPageProvider({ getService, getPageObjects }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+  const PageObjects = getPageObjects(['common']);
+
+  const getColumnIndex = async (tableTestId: string, columnName: string) => {
+    const table = await testSubjects.find(tableTestId);
+    const headers = await table.findAllByCssSelector('thead th');
+    const headerIndexes = await Promise.all(headers.map((header) => header.getVisibleText()));
+    const columnIndex = headerIndexes.findIndex((i) => i === columnName);
+    return [columnIndex, headers[columnIndex]] as [
+      number,
+      Awaited<ReturnType<typeof testSubjects.find>>
+    ];
+  };
+
+  const getFilterColumnValues = async (tableTestId: string, columnName: string) => {
```

Comment:
seems it doesn't return the correct values

Reply:
fixed this and added an assertion to verify we're not operating on empty values, which ends up a false positive. thank you for catching this!